### DISCUSSION TITLE
Would it be good to add eventHappenOn field in all events

### DISCUSSION THREAD

**Discussion by shashikantpawar21** [Eventuous/eventuous#113] [Ideas]
In most of the scenario, I see we can use eventHappenOn Time to do some logic work.. Else we have to add a dateTime field to keep track of when did the event happen and to use that value in projection..

Alexey... I know i may be asking too many question or suggestion.. 
I would like to take up some incremental little work in eventuous as open source contribution.. if you have some I would be happy to do it  in my free time

**Comment by alexeyzimarev** [comment]
I don't like the idea of having a generic thing like `EventHappenedOn`, but you can always create an abstract record or class and inherit your events from it. Set the default value to `DateTime.Now`, no problem. I prefer to have properties with meaningful names.
